Index: application/views/admin/admin_products_dashboard_page.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/application/views/admin/admin_products_dashboard_page.php b/application/views/admin/admin_products_dashboard_page.php
--- a/application/views/admin/admin_products_dashboard_page.php	
+++ b/application/views/admin/admin_products_dashboard_page.php	(date 1676801943785)
@@ -10,9 +10,10 @@
     <link rel="stylesheet" type="text/css" href="<?= base_url('assets/css/normalize.css') ?>" />
     <link rel="stylesheet" type="text/css" href="<?= base_url('assets/css/style.css') ?>" />
     <script>
-        $.get('products/get_products', function (res) {
+        $.get('../products/get_products', function (res) {
+			/*  Open edit product dialog box    */
 			$('#products').html(res);
-        })
+        });
         /*  For pagination highlight    */
         function pageNumHighlight(pageNum){
             $(".pagination > a").css("background-color", "white").css("color", "blue");
@@ -163,124 +164,183 @@
             });
             /**********************************************/
 
-             /*  Open edit product dialog box    */
-             $(document).on("click", ".product_edit_link", function(){
-                var productID = $(this).parent().parent().siblings(".product_id").text();
-                var headerStr = "Edit Product - ID " + productID;
-                var productName = $(this).parent().parent().siblings(".product_id + td").text();
-                var productDesc = "Product description...Product description...Product description...Product description...Product description...Product description...Product description...";
-                var productCategory = productName;
-                var productPrice = 19.99;
-                var productInventory = $(this).parent().parent().siblings(".product_id + td + td").text();
-                var productImgSrc = $(this).parent().parent().parent().children("td:first-child").find("img").attr("src");
-                var productImgAlt = $(this).parent().parent().parent().children("td:first-child").find("img").attr("alt");
+			/*  Open edit product dialog box    */
+			$(document).on("click", ".product_edit_link", function(){
+				$.get($(this).attr('href'), function (res) {
+					data = $.parseJSON(res);
+
+					var productID = $(this).parent().parent().siblings(".product_id").text();
+					var headerStr = "Edit Product - ID " + productID;
+					var productName = data.product['name'];
+					var productDesc = data.product['description'];
+					var productCategory = data.product['category_name'];
+					var productPrice = data.product['price'];
+					var productInventory = data.product['stock'];
+						// $(this).parent().parent().siblings(".product_id + td + td").text();
+					var productImgSrc = $(this).parent().parent().parent().children("td:first-child").find("img").attr("src");
+					var productImgAlt = $(this).parent().parent().parent().children("td:first-child").find("img").attr("alt");
 
-                var htmlImgStr = "" +
-                    '<li class="img_upload_section">' +
-                        '<figure>' +
-                            '<img src="' + productImgSrc + '" alt="' + productImgAlt + '" />' +
-                        '</figure>' +
-                        '<p class="img_filename">' + productImgAlt + '</p>' +
-                        '<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-trash btn_img_upload_delete" viewBox="0 0 16 16">' +
-                            '<path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"/>' +
-                            '<path fill-rule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"/>' +
-                        '</svg>' +
-                        // '<input type="checkbox" name="is_img_upload_main_id" value="filename" />' +
-                        '<input type="checkbox" name="img_upload_main_id" value="filename" />' +
-                        '<label>main</label>' +
-                    '</li>';
+					var htmlImgStr = "" +
+						'<li class="img_upload_section">' +
+						'<figure>' +
+						'<img src="' + productImgSrc + '" alt="' + productImgAlt + '" />' +
+						'</figure>' +
+						'<p class="img_filename">' + productImgAlt + '</p>' +
+						'<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-trash btn_img_upload_delete" viewBox="0 0 16 16">' +
+						'<path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"/>' +
+						'<path fill-rule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"/>' +
+						'</svg>' +
+						// '<input type="checkbox" name="is_img_upload_main_id" value="filename" />' +
+						'<input type="checkbox" name="img_upload_main_id" value="filename" />' +
+						'<label>main</label>' +
+						'</li>'
+					;
 
-                $(".add_edit_product_header").text(headerStr);
-                $(".input_product_name").val(productName);
-                $(".input_product_desc").val(productDesc);
-                $(".dummy_select_tag span:first-child").text(productCategory);
-                $(".input_product_price").val(productPrice);
-                $(".input_product_qty").val(productInventory);
-                $(".img_upload_container").html(htmlImgStr);
+					$(".add_edit_product_header").text(headerStr);
+					$(".input_product_name").val(productName);
+					$(".input_product_desc").val(productDesc);
+					$(".dummy_select_tag span:first-child").text(productCategory);
+					$(".input_product_price").val(productPrice);
+					$(".input_product_qty").val(productInventory);
+					$(".img_upload_container").html(htmlImgStr);
 
-                $(".products_add_edit_btn .product_id").val(productID);
-                $(".btn_submit_products_add_edit").val("Update");
-                $(".btn_submit_products_add_edit").removeClass("add_product_submit");
-                $(".btn_submit_products_add_edit").addClass("edit_product_submit");
-                $(".modal_bg").show();
-                $("dialog.admin_products_add_edit").show();
-                return false;
-            });
-            /**********************************************/
-
-            /*  Clicking add button will submit the form using ajax    */
-            // submit form using the general ajax. Not this!
-            $(document).on("click", ".edit_product_submit", function(){
-                var productIdEdited = ".product_id_" + $(".products_add_edit_btn .product_id").val();
-                var productName = $(".input_product_name").val();
-                var productInventory = $(".input_product_qty").val();
-
-                var imgUpload = $(".img_upload_section > figure > img");
-                var imgCheckbox = $(".img_upload_section > input[type=checkbox]");
-                var prevProductImg = [];
-                var mainIndexImg = 0;
-                for(var i = 0; i < imgUpload.length; i++){
-                    prevProductImg[i] = imgUpload[i].currentSrc;
-                    if(imgCheckbox[i].checked){
-                        mainIndexImg = i;
-                    }
-                }
-                var productImgSrc = prevProductImg[mainIndexImg];
-                var productImgAlt = "img";
-
-                $(productIdEdited).children(".product_id + td").text(productName);
-                $(productIdEdited).children(".product_id + td + td").text(productInventory);
-                $(productIdEdited).children("td:first-child").find("img").attr("src", productImgSrc);
-                $(productIdEdited).children("td:first-child").find("img").attr("alt", productImgAlt);
-                $(this).parent().parent().submit(function(){ return false; });
-                hideDialogBox();
-                return false;
-            });
-            /**********************************************/
-
+					$(".products_add_edit_btn .product_id").val(productID);
+					$(".btn_submit_products_add_edit").val("Update");
+					$(".btn_submit_products_add_edit").removeClass("add_product_submit");
+					$(".btn_submit_products_add_edit").addClass("edit_product_submit");
+					$(".modal_bg").show();
+					$("dialog.admin_products_add_edit").show();
 
-            /*  Initializing the content of product categories    */
-            var categories = ["Shirt", "Hat", "Mug", "Pant", "Key Chain", "Belt"];
-            var categoriesOption = "<form></form>";
-            var selectOptions = "";
-            for(var i = 0; i < categories.length; i++){
-                categoriesOption += 
-                    '<li class="product_category_edit_delete_section arr_' + i + '">' +
+					/*  Initializing the content of product categories    */
+					var categories = ["Shirt", "Hat", "Mug", "Pant", "Key Chain", "Belt"];
+					// var categories = [];
+					// for (let key in data.categories) {
+					// 	categories[key] = data.categories[key]['name'];
+					// }
+					var categoriesOption = "<form></form>";
+					var selectOptions = "";
+					for(var i = 0; i < categories.length; i++){
+						for (let key in data.categories) {
+							if (key == i) {
+								category_id = data.categories[key]['id'];
+								console.log(category_id);
+							}
+						}
+						categoriesOption +=
+							'<li class="product_category_edit_delete_section arr_' + i + '">' +
 
-                        '\n\t<form class="form_product_category_edit" action="" method="post">' +
-                            '\n\t\t<input class="product_category_id" type="hidden" name="product_category_id" value="arr_' + i + '"/>' +
-                            '\n\t\t<input class="product_category_text_input" readonly type="text" value="' + categories[i] + '"/>' +
-                        '\n\t</form>' +
+							'\n\t<form class="form_product_category_edit" action="" method="post">' +
+							'\n\t\t<input class="product_category_id" type="hidden" name="product_category_id" value="' + category_id + '"/>' +
+							'\n\t\t<input class="product_category_text_input" readonly type="text" value="' + categories[i] + '"/>' +
+							'\n\t</form>' +
 
-                        '\n\t<div class="product_category_btn">' +
+							'\n\t<div class="product_category_btn">' +
 
-                            '\n\t\t<div class="waiting_icon"><img src="../../../application/views/assets/img/ajax-loading-icon.gif" alt="waiting icon"></div>' +
+							'\n\t\t<div class="waiting_icon"><img src="../../../application/views/assets/img/ajax-loading-icon.gif" alt="waiting icon"></div>' +
 
-                            '\n\t\t<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-pencil-fill btn_product_category_edit" viewBox="0 0 16 16">' +
-                                '\n\t\t\t<path d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"/>' +
-                            '\n\t\t</svg>' +
+							'\n\t\t<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-pencil-fill btn_product_category_edit" viewBox="0 0 16 16">' +
+							'\n\t\t\t<path d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"/>' +
+							'\n\t\t</svg>' +
 
-                            '\n\t\t<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-trash btn_product_category_delete" viewBox="0 0 16 16">' +
-                                '\n\t\t\t<path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"/>' +
-                                '\n\t\t\t<path fill-rule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"/>' +
-                            '\n\t\t</svg>' +
+							'\n\t\t<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-trash btn_product_category_delete" viewBox="0 0 16 16">' +
+							'\n\t\t\t<path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"/>' +
+							'\n\t\t\t<path fill-rule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"/>' +
+							'\n\t\t</svg>' +
 
-                        '\n\t</div>' +
+							'\n\t</div>' +
 
-                    '\n</li>';
+							'\n</li>';
 
-                selectOptions += '<option value="' + categories[i] + '">' + categories[i] + '</option>';
-            }
-            $(".product_categories").html(categoriesOption);
-            resetCategoryDisplay();
+						selectOptions += '<option value="' + categories[i] + '">' + categories[i] + '</option>';
+					}
+					$(".product_categories").html(categoriesOption);
+					resetCategoryDisplay();
 
-            /**********************************************/
+					/**********************************************/
 
-            /*  Show the options/categories for the dummy select tag    */
-            $(document).on("click", ".dummy_select_tag", function(){
-                $(this).css("border", "2px solid black");
-                $(".product_categories").toggle();
+					/*  Show the options/categories for the dummy select tag    */
+					$(document).on("click", ".dummy_select_tag", function(){
+						$(this).css("border", "2px solid black");
+						$(".product_categories").toggle();
+					});
+				});
+				return false;
+			});
+            /**********************************************/
+
+            /*  Clicking add button will submit the form using ajax    */
+            // submit form using the general ajax. Not this!
+            $(document).on("click", ".edit_product_submit", function(){
+                var productIdEdited = ".product_id_" + $(".products_add_edit_btn .product_id").val();
+                var productName = $(".input_product_name").val();
+                var productInventory = $(".input_product_qty").val();
+
+                var imgUpload = $(".img_upload_section > figure > img");
+                var imgCheckbox = $(".img_upload_section > input[type=checkbox]");
+                var prevProductImg = [];
+                var mainIndexImg = 0;
+                for(var i = 0; i < imgUpload.length; i++){
+                    prevProductImg[i] = imgUpload[i].currentSrc;
+                    if(imgCheckbox[i].checked){
+                        mainIndexImg = i;
+                    }
+                }
+                var productImgSrc = prevProductImg[mainIndexImg];
+                var productImgAlt = "img";
+
+                $(productIdEdited).children(".product_id + td").text(productName);
+                $(productIdEdited).children(".product_id + td + td").text(productInventory);
+                $(productIdEdited).children("td:first-child").find("img").attr("src", productImgSrc);
+                $(productIdEdited).children("td:first-child").find("img").attr("alt", productImgAlt);
+                $(this).parent().parent().submit(function(){ return false; });
+                hideDialogBox();
+                return false;
             });
+            /**********************************************/
+
+
+			// /*  Initializing the content of product categories    */
+			// var categories = ["Shirt", "Hat", "Mug", "Pant", "Key Chain", "Belt"];
+			// var categoriesOption = "<form></form>";
+			// var selectOptions = "";
+			// for(var i = 0; i < categories.length; i++){
+			// 	categoriesOption +=
+			// 		'<li class="product_category_edit_delete_section arr_' + i + '">' +
+			//
+			// 		'\n\t<form class="form_product_category_edit" action="" method="post">' +
+			// 		'\n\t\t<input class="product_category_id" type="hidden" name="product_category_id" value="arr_' + i + '"/>' +
+			// 		'\n\t\t<input class="product_category_text_input" readonly type="text" value="' + categories[i] + '"/>' +
+			// 		'\n\t</form>' +
+			//
+			// 		'\n\t<div class="product_category_btn">' +
+			//
+			// 		'\n\t\t<div class="waiting_icon"><img src="../../../application/views/assets/img/ajax-loading-icon.gif" alt="waiting icon"></div>' +
+			//
+			// 		'\n\t\t<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-pencil-fill btn_product_category_edit" viewBox="0 0 16 16">' +
+			// 		'\n\t\t\t<path d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"/>' +
+			// 		'\n\t\t</svg>' +
+			//
+			// 		'\n\t\t<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-trash btn_product_category_delete" viewBox="0 0 16 16">' +
+			// 		'\n\t\t\t<path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"/>' +
+			// 		'\n\t\t\t<path fill-rule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"/>' +
+			// 		'\n\t\t</svg>' +
+			//
+			// 		'\n\t</div>' +
+			//
+			// 		'\n</li>';
+			//
+			// 	selectOptions += '<option value="' + categories[i] + '">' + categories[i] + '</option>';
+			// }
+            // $(".product_categories").html(categoriesOption);
+            // resetCategoryDisplay();
+			//
+            // /**********************************************/
+			//
+            // /*  Show the options/categories for the dummy select tag    */
+            // $(document).on("click", ".dummy_select_tag", function(){
+            //     $(this).css("border", "2px solid black");
+            //     $(".product_categories").toggle();
+            // });
             /**********************************************/
 
              /*  Show the edit/delete buttons on hover.    */
